native/kotlin/api/kotlin/src/main/kotlin/rs/wordpress/api/kotlin/WpRequestExecutor.kt

```diff
@@ -85,54 +84,57 @@ class WpRequestExecutor(
             }
         }
 
-    override suspend fun uploadMedia(mediaUploadRequest: MediaUploadRequest): WpNetworkResponse =
+    override suspend fun upload(request: WpMultipartFormRequest): WpNetworkResponse =
         withContext(dispatcher) {
-            val requestBuilder = Request.Builder().url(mediaUploadRequest.url())
+            val requestBuilder = Request.Builder().url(request.url())
             val multipartBodyBuilder = MultipartBody.Builder()
                 .setType(MultipartBody.FORM)
-            mediaUploadRequest.mediaParams().forEach { (k, v) ->
+            request.fields().forEach { (k, v) ->
                 multipartBodyBuilder.addFormDataPart(k, v)
             }
-            val file = fileResolver.getFile(mediaUploadRequest.filePath())
-            if (file == null || !file.canBeUploaded()) {
-                throw MediaUploadRequestExecutionException.MediaFileNotFound(mediaUploadRequest.filePath())
+            request.files().forEach { (name, fileInfo) ->
+                val file = fileResolver.getFile(fileInfo.filePath)
+                if (file == null || !file.canBeUploaded()) {
+                    throw RequestExecutionException.MediaFileNotFound(filePath = fileInfo.filePath)
+                }
+                val mimeType = fileInfo.mimeType ?: "application/octet-stream"
+                val requestBody = getRequestBody(file, mimeType, uploadListener)
```

Comment:
@oguzkocer I'm not sure if the progress reporting is still working as expected. The previous implementation only supported uploading one file, but the new implementation supports uploading many files. I'm not sure if the `uploadListener` still works.